### PR TITLE
oauth2-scopes-with-cxf-redirection

### PR DESCRIPTION
**Pull request by topolik**

### CODE REVIEW

File: modules/apps/oauth2-provider/oauth2-provider-scopes-impl/src/main/java/com/liferay/oauth2/provider/scopes/impl/ThreadLocalServiceScopeChecker.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+		return true;
+	}
+
+	@Override
+	public boolean hasAnyScope(String... scopes) {
+		if (Validator.isNull(scopes)) {
+			throw new IllegalArgumentException("Scopes can't be null");
+		}
+		Collection<OAuth2ScopeGrant> oAuth2ScopeGrants =
+			_oAuth2ScopeGrantLocalService.findByA_BNS_BV_C_T(
```
[the remark this answers is not in the document]
Comment (csierra) [reply modules/apps/oauth2-provider/oauth2-provider-scopes-impl/src/main/java/com/liferay/oauth2/provider/scopes/impl/ThreadLocalServiceScopeChecker.java]: aaah... dyslexia

File: modules/apps/oauth2-provider/oauth2-provider-scopes-impl/src/main/java/com/liferay/oauth2/provider/scopes/impl/cxf/LiferayAuthorizationCodeGrantHandlerRegistrator.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			AuthorizationCodeGrantHandler authorizationCodeGrantHandler =
+				new AuthorizationCodeGrantHandler();
+
+			authorizationCodeGrantHandler.setDataProvider(
+				_liferayOAuthDataProvider);
+
+			_serviceRegistration = bundleContext.registerService(
+				AccessTokenGrantHandler.class,
+				authorizationCodeGrantHandler, new Hashtable<>());
+		}
```
[the remark this answers is not in the document]
Comment (csierra) [reply modules/apps/oauth2-provider/oauth2-provider-scopes-impl/src/main/java/com/liferay/oauth2/provider/scopes/impl/cxf/LiferayAuthorizationCodeGrantHandlerRegistrator.java]: oh... definitely we can

File: modules/apps/oauth2-provider/oauth2-provider-scopes-impl/src/main/java/com/liferay/oauth2/provider/scopes/impl/cxf/OAuth2AuthVerifier.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		String authorization = request.getHeader(HttpHeaders.AUTHORIZATION);
+
+		AuthVerifierResult authVerifierResult = new AuthVerifierResult();
+
+		try {
+			String[] basicAuthParts = authorization.split(" ");
+
+			String basicAuthPart = basicAuthParts[0];
+
+			if (!"Bearer".equalsIgnoreCase(basicAuthPart)) {
```
[the remark this answers is not in the document]
Comment (csierra) [reply modules/apps/oauth2-provider/oauth2-provider-scopes-impl/src/main/java/com/liferay/oauth2/provider/scopes/impl/cxf/OAuth2AuthVerifier.java]: the token retriever is needed as a ways to retrieve the token for JAX-RS only

File: modules/apps/oauth2-provider/oauth2-provider-scopes-impl/src/main/java/com/liferay/oauth2/provider/scopes/impl/cxf/OAuth2AuthVerifier.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+			if (accessToken == null) {
+				return new AuthVerifierResult();
+			}
+
+			_scopeContext.setTokenString(token);
+			
+			for (OAuthPermission scope : accessToken.getScopes()) {
+				ServiceAccessPolicyThreadLocal.addActiveServiceAccessPolicyName(
+					"OAUTH_" + scope.getPermission());
```
[the remark this answers is not in the document]
Comment (csierra) [reply modules/apps/oauth2-provider/oauth2-provider-scopes-impl/src/main/java/com/liferay/oauth2/provider/scopes/impl/cxf/OAuth2AuthVerifier.java]: which scopes would we pass?

[the remark this answers is not in the document]
Comment (topolik) [reply modules/apps/oauth2-provider/oauth2-provider-scopes-impl/src/main/java/com/liferay/oauth2/provider/scopes/impl/cxf/OAuth2AuthVerifier.java]: I need to try to see

**Pull request closed without merging**